compiler/rustc_middle/src/thir.rs

```diff
@@ -1161,6 +1167,9 @@ impl<'tcx> fmt::Display for Pat<'tcx> {
                 }
                 write!(f, "{subpattern}")
             }
+            PatKind::DerefPattern { ref subpattern } => {
+                write!(f, "box {subpattern}")
```

Comment:
```suggestion
                write!(f, "k#deref {subpattern}")
```

i know it's not syntactical, but i'd rather make this a distinct pattern constructor in the printed THIR